solidity/dashboard/src/services/liquidity-rewards.js

```diff
@@ -159,6 +163,46 @@ class UniswapLPRewards extends LiquidityRewards {
 
     return this._calculateAPY(r, WEEKS_IN_YEAR)
   }
+
+  /**
+   * Calculates lp token balance for the given pair
+   * The calculations were done based on
+   * https://uniswap.org/docs/v2/advanced-topics/understanding-returns/#why-is-my-liquidity-worth-less-than-i-put-in
+   *
+   * @param {string} lpBalance Balance of liquidity token for a given uniswap pair deposited in
+   * the LPRewards` contract.
+   * @return {Promise<{token0: string, token1: string}>}
+   */
+  calculateLPTokenBalance = async (lpBalance) => {
+    const pairData = await getPairData(this.wrappedTokenAddress.toLowerCase())
+
+    const shareOfUniswapPool = percentageOf(
+      lpBalance.toString(),
+      fromTokenUnit(pairData.totalSupply).toString()
+    )
+
+    const constantProduct = new BigNumber(pairData.reserve0).multipliedBy(
+      pairData.reserve1
+    )
+
+    const token0LiquidityPool = constantProduct
+      .div(pairData.token1Price)
+      .squareRoot()
+    const token1LiquidityPool = constantProduct
+      .multipliedBy(pairData.token1Price)
+      .squareRoot()
+
```

Comment:
@michalsmiarowski @r-czajkowski guys, are you sure we need all these? What we have is:
`token_liquidity_pool` = `pairData.reserve0`, where token is ex. KEEP
`eth_liquidity_pool` = `pairData.reserve1`

```
const eth_price = token_liquidity_pool / eth_liquidity_pool // KEEP price for 1 ETH
const constantProduct = pairData.reserve0 * pairData.reserve1 => token_liquidity_pool * eth_liquidity_pool

const token1LiquidityPool = (token_liquidity_pool * eth_liquidity_pool / (token_liquidity_pool / eth_liquidity_pool)).squareRoot() => token1LiquidityPool = eth_liquidity_pool, which is the same.
```
It seems that it should be enough to write:
```
return {
      token0: fromPercentage(
        shareOfUniswapPool,
        pairData.reserve0
      ).toString(),
      token1: fromPercentage(
        shareOfUniswapPool,
        pairData.reserve1
      ).toString(),
    }
```

What do you think?

Reply:
@dimpar yes, that's a good point. I tested it and everything works wonderfully.

I think that we can go even further and remove the shareOfThePool calculation and just do:

```
return {
    token0: fromTokenUnit(
      new BigNumber(lpBalance)
        .multipliedBy(pairData.reserve0)
        .dividedBy(pairData.totalSupply)
    ).toString(),
    token1: fromTokenUnit(
      new BigNumber(lpBalance)
        .multipliedBy(pairData.reserve1)
        .dividedBy(pairData.totalSupply)
    ).toString(),
  }
```

becuase right now we multiply the result by 100 when calling `percentageOf` function, and then in return we divide the result by 100 by calling `fromPercentage` function. So it's just unnecessary and can be simplified as above.

Same calculation can be found in [uniswap-sdk repo](https://github.com/Uniswap/uniswap-sdk/blob/a88048e9c4198a5bdaea00883ca00c8c8e582605/src/entities/pair.ts#L225).

Reply:
yup, it can be simplified further as you described @michalsmiarowski